### DISCUSSION TITLE
How can i join 2 tables on multiple columns

### DISCUSSION THREAD

**Discussion by ypogeios** [ujjwalguptaofficial/JsStore#359] [Q&A]
Hello,
How can i join 2 tables on multiple columns?
For example
var results = await jsstoreCon.select({
            from: "Order_Details",
            join:{
                with:"Order_Details_Options",
                on: 'Order_Details.order_id=Order_Details_Options.order_id AND Order_Details.id=Order_Details_Options.order_details_line_id',
                as: {
                    id: "DetailsID",
                    name:"DetailsName",
                    product_id:"DetailsProdID"
                }
            }
            });

**Comment by ujjwalguptaofficial** [comment]
Try multiple joins like this - 

```
select({
    from: 'Orders',
    join: [{
            with: 'Customers',
            on: "Orders.customerId=Customers.customerId",
            as: {
                customerId: 'cId'
            }
        },
        {
            with: 'Customers',
            on: "Orders.orderId=Customers.customerId",
            as: {
                customerId: 'cId1'
            }
        }
    ]
});
```

One column join in first join and in second column join in second join query. 

Let me know if this does not work.
